test/IISIntegration.FunctionalTests/Inprocess/StartupTests.cs

```diff
@@ -2,216 +2,116 @@
-                    {
-                        Assert.Equal("Hello World", responseText);
-                    }
-                    catch (XunitException)
-                    {
-                        logger.LogWarning(response.ToString());
-                        logger.LogWarning(responseText);
-                        throw;
-                    }
-                }
-            }
+            var deploymentParameters = GetBaseDeploymentParameters();
+
+            // Point to dotnet installed in user profile.
+            deploymentParameters.EnvironmentVariables["DotnetPath"] = dotnetLocation;
+
+            var deploymentResult = await DeployAsync(deploymentParameters);
+
+            ModifyAspNetCoreSectionInWebConfig(deploymentResult, "processPath", "%DotnetPath%");
+
+            var response = await deploymentResult.RetryingHttpClient.GetAsync("HelloWorld");
+
+            var responseText = await response.Content.ReadAsStringAsync();
+            Assert.Equal("Hello World", responseText);
         }
 
         [Fact]
         public async Task InvalidProcessPath_ExpectServerError()
         {
             var dotnetLocation = "bogus";
-            using (StartLog(out var loggerFactory))
-            {
-                var logger = loggerFactory.CreateLogger("HelloWorldTest");
-                var deploymentParameters = GetBaseDeploymentParameters();
 
-                // Point to dotnet installed in user profile.
-                deploymentParameters.EnvironmentVariables["DotnetPath"] = Environment.ExpandEnvironmentVariables(dotnetLocation); // Path to dotnet.
+            var deploymentParameters = GetBaseDeploymentParameters();
+            // Point to dotnet installed in user profile.
+            deploymentParameters.EnvironmentVariables["DotnetPath"] = Environment.ExpandEnvironmentVariables(dotnetLocation); // Path to dotnet.
 
-                using (var deployer = ApplicationDeployerFactory.Create(deploymentParameters, loggerFactory))
-                {
-                    var deploymentResult = await deployer.DeployAsync();
+            var deploymentResult = await DeployAsync(deploymentParameters);
 
-                    Helpers.ModifyAspNetCoreSectionInWebConfig(deploymentResult, "processPath", "%DotnetPath%");
+            ModifyAspNetCoreSectionInWebConfig(deploymentResult, "processPath", "%DotnetPath%");
 
-                    // Request to base address and check if various parts of the body are rendered & measure the cold startup time.
-                    var response = await RetryHelper.RetryRequest(() =>
-                    {
-                        return deploymentResult.HttpClient.GetAsync("HelloWorld");
-                    }, logger, deploymentResult.HostShutdownToken, retryCount: 30);
+            // Request to base address and check if various parts of the body are rendered & measure the cold startup time.
+            var response = await deploymentResult.RetryingHttpClient.GetAsync("HelloWorld");
```

Comment:
Using the retry handler when the expected result is a InternalServerError is a little odd. Not a new issue.

Reply:
We want to get 500 but if we happen to get 503 because the server is starting retrying is OK.